database/redis/last_check.go

```diff
@@ -97,47 +97,38 @@ func (connector *DbConnector) SetTriggerCheckMaintenance(triggerID string, metri
 	var readingErr error
 
 	lastCheckString, readingErr := c.Get(ctx, metricLastCheckKey(triggerID)).Result()
-	if readingErr != nil && readingErr != redis.Nil {
-		return readingErr
+	if readingErr != nil {
+		if readingErr != redis.Nil {
+			return readingErr
+		}
+		return nil
 	}
 
-	for readingErr != redis.Nil {
-		var lastCheck = moira.CheckData{}
-		err := json.Unmarshal([]byte(lastCheckString), &lastCheck)
-		if err != nil {
-			return fmt.Errorf("failed to parse lastCheck json %s: %s", lastCheckString, err.Error())
-		}
-		metricsCheck := lastCheck.Metrics
-		if len(metricsCheck) > 0 {
-			for metric, value := range metrics {
-				data, ok := metricsCheck[metric]
-				if !ok {
-					continue
-				}
-				moira.SetMaintenanceUserAndTime(&data, value, userLogin, timeCallMaintenance)
-				metricsCheck[metric] = data
+	var lastCheck = moira.CheckData{}
+	err := json.Unmarshal([]byte(lastCheckString), &lastCheck)
+	if err != nil {
+		return fmt.Errorf("failed to parse lastCheck json %s: %s", lastCheckString, err.Error())
+	}
+	metricsCheck := lastCheck.Metrics
+	if len(metricsCheck) > 0 {
+		for metric, value := range metrics {
+			data, ok := metricsCheck[metric]
+			if !ok {
+				continue
 			}
+			moira.SetMaintenanceUserAndTime(&data, value, userLogin, timeCallMaintenance)
+			metricsCheck[metric] = data
 		}
-		if triggerMaintenance != nil {
-			moira.SetMaintenanceUserAndTime(&lastCheck, *triggerMaintenance, userLogin, timeCallMaintenance)
-		}
-		newLastCheck, err := json.Marshal(lastCheck)
-		if err != nil {
-			return err
-		}
-
-		var prev string
-		prev, readingErr = c.GetSet(ctx, metricLastCheckKey(triggerID), newLastCheck).Result()
```

Comment:
Is the second line of the comment for function

> If during the update lastCheck was updated from another place, try update again

still relevant?

Reply:
Good catch, removed